packages/expo-polyfills/src/index.ts

```diff
@@ -176,42 +176,46 @@ class SyncSecureStorage implements Storage {
 
   constructor() {
     this.storage = new Map<string, string>();
-    this.initPromise = new Promise((resolve, reject) => {
-      // Fetch ::keys::
-      SecureStore.getItemAsync('___keys___')
-        .then((keysStr) => {
-          if (!keysStr) {
-            // No keys currently, just resolve
-            this.initialized = true;
-            resolve();
-            return;
-          }
-          // Parse keys
-          const keys = JSON.parse(keysStr) as string[];
-          const promises = keys.map((key) => {
-            return new Promise<[string, string | null]>((resolve, reject) => {
-              SecureStore.getItemAsync(key)
-                .then((val) => {
-                  resolve([key, val]);
-                })
-                .catch(reject);
-            });
-          });
-          Promise.all(promises)
-            .then((values) => {
-              for (const [key, value] of values) {
-                if (!value) {
-                  continue;
-                }
-                this.storage.set(key, value);
-              }
-              this.initialized = true;
-              resolve();
-            })
-            .catch(reject);
-        })
-        .catch(reject);
-    });
+    this.initPromise = this.initStorage();
+  }
+
+  private async initStorage(): Promise<void> {
+    let keysStr: string | null;
+    try {
+      // We can't just get all the "keys" that currently exist as there is no such concept
+      // We need to instead store all keys we intend to restore later in a known key
+      // We've chose `___keys___` to prevent collisions
+      // Here we get the keys and attempt to restore them
+      keysStr = await SecureStore.getItemAsync('___keys___');
+    } catch (err: unknown) {
+      // If an error is thrown here, it's likely an issue with encryption...
+      // We may be unable to decrypt existing keys if the keystore itself has changed
+      // In that case we should dump the keys and start fresh
+      // We can just initialize and resolve the `initPromise`
+      console.error(err);
+      await SecureStore.deleteItemAsync('___keys___');
```

Comment:
Do we need to wrap this in `try`/`catch` too?

At least now that the method is `async`, if this fails the promise will reject, so it's not as urgent.

Reply:
Yeah I think if delete fails we probably want it to reject the promise. Something is probably wrong in that case